app/src/main/java/com/hover/stax/data/repository/AccountRepositoryImpl.kt

```diff
@@ -22,30 +24,34 @@ import org.json.JSONObject
 import org.koin.core.component.KoinComponent
 import org.koin.core.component.inject
 
-class AccountRepositoryImpl(val accountRepo: AccountRepo, val channelRepo: ChannelRepo, val actionRepo: ActionRepo, private val coroutineDispatcher: CoroutineDispatcher) : AccountRepository, PushNotificationTopicsInterface, KoinComponent {
+class AccountRepositoryImpl(val accountRepo: AccountRepo, val channelRepo: ChannelRepo, private val presentSimUseCase: GetPresentSimUseCase,  val actionRepo: ActionRepo, private val coroutineDispatcher: CoroutineDispatcher) : AccountRepository, PushNotificationTopicsInterface, KoinComponent {
```

Comment:
You should use the repo not the usecase here. UseCases should be used only on the presentation layer. While testing, we provide mocks of the repos used.

Reply:
Thanks for this.